File: ProjectEuler/051-100/S57.cs
using System.Numerics;

namespace ProjectEuler
{
    public class S57 : ISolution
    {
        public string GetAnswer()
        {
            int count = 0;

            BigInteger denominator = 1;
            BigInteger numerator = 1;
            for (int i = 0; i < 1000; i++)
            {
                BigInteger a = denominator;
                BigInteger b = numerator;
                denominator = a + b;
                numerator = 2 * a + b;
                if (numerator.ToString().Length > denominator.ToString().Length)
                {
                    count++;
                }
            }

            return count.ToString();
        }
    }
}

File: src/051-100/S57.cc
#include "S57.h"
#include "../Utils.h"

using namespace std;

namespace ProjectEuler
{
std::string S57::GetAnswer()
{
	int count = 0;
	BigInt num(3);
	BigInt den(2);

	for (int i = 1; i <= 1000; i++)
	{
		if (num.toString().length() > den.toString().length())
		{
			count++;
		}

		BigInt nextNum = num + den + den;
		BigInt nextDen = num + den;
		num = nextNum;
		den = nextDen;
	}
	return std::to_string(count);
}
} // namespace ProjectEuler
